### PR TITLE
qsvenc: support the combination of LA and CBR

### PR DESCRIPTION
**Pull request by guanjunz**
add LA CBR supports

### CODE REVIEW

File: libavcodec/qsvenc.c [hunk 1]
```diff
@@ -756,6 +756,13 @@ static int init_video_param(AVCodecContext *avctx, QSVEncContext *q)
         q->param.mfx.TargetKbps  = target_bitrate_kbps / brc_param_multiplier;
         q->extco2.LookAheadDepth = q->look_ahead_depth;
         q->param.mfx.BRCParamMultiplier = brc_param_multiplier;
+        q->param.mfx.BufferSizeInKB = buffer_size_in_kilobytes / brc_param_multiplier;
+        q->param.mfx.InitialDelayInKB = initial_delay_in_kilobytes / brc_param_multiplier;
+        q->param.mfx.MaxKbps = max_bitrate_kbps / brc_param_multiplier;
+        if (avctx->bit_rate == avctx->rc_max_rate)
+        {
+            q->param.mfx.RateControlMethod = MFX_RATECONTROL_CBR;
```
Comment (xhaihao) [inline libavcodec/qsvenc.c]: Is there a way to fix this in select_rc_mode() ?

**Comment by xhaihao** [comment]
You changed qsvenc.c only, so please use 'qsvenc:' instead of 'qsvenc_{hevc,h264}' in the log

File: libavcodec/qsvenc.c (hunk 1 above)
Comment (guanjunz) [reply libavcodec/qsvenc.c]: I think what you suggest is more reasonable. There is a problem to handle 
if (q->extbrc) {
            q->extco2.LookAheadDepth = q->look_ahead_depth;
        }
and 
LA what we need

**Comment by xhaihao** [comment]
@guanjunz Is this PR still needed ? If yes, could you please update your patch ?

**Pull request closed without merging**